2017-03-01-preview/Microsoft.Insights.ActivityLogAlert.json

```diff
@@ -0,0 +1,199 @@
+          ],
+          "description": "Indicates whether this activity log alert is enabled. If an activity log alert is not enabled, then none of its actions will be activated."
+        },
+        "condition": {
+          "oneOf": [
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            },
+            {
+              "$ref": "/definitions/ActivityLogAlertAllOfCondition"
+            }
+          ],
+          "description": "The conditon that will cause this alert to activate."
+        },
+        "actions": {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/ActivityLogAlertActionList"
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "The actions that will activate when the condition is met."
+        },
+        "description": {
+          "type": "string",
+          "minLength": 1,
+          "description": "A description of this activity log alert."
+        }
+      },
+      "required": [
+        "scopes",
+        "enabled",
+        "condition",
+        "actions"
+      ],
+      "description": "An Azure Activity log alert."
+    },
+    "ActivityLogAlertActionGroup": {
+      "type": "object",
+      "properties": {
+        "actionGroupId": {
+          "type": "string",
+          "description": "The resourceId of the action group."
+        },
+        "webhookPropertios": {
+          "oneOf": [
+            {
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              }
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "the dictionary of custom properties to include with the post operation. These data are appended to the webhook payload."
+        }
+      },
+      "required": [
+        "actionGroupId"
+      ],
+      "description": "A pointer to an Azure Action Group."
+    },
+    "ActivityLogAlertActionList": {
+      "type": "object",
+      "properties": {
+        "actionGroups": {
+          "oneOf": [
+            {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/ActivityLogAlertActionGroup"
+              }
+            },
+            {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+            }
+          ],
+          "description": "The list of activity log alerts."
+        }
+      },
+      "description": "A list of activity log alert actions."
+    },
+    "ActivityLogAlertAllOfCondition": {
+      "type": "object",
+      "properties": {
+        "allOf": {
```

Comment:
We may want to test this out. "allOf" is a keyword for JSON schema and here it's used as property name.